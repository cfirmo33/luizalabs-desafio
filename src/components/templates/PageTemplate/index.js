import React from 'react'
import PropTypes from 'prop-types'
import Header from '../../Layout/Header'

const PageTemplate = ({ children }) => (
  <div className="container">
    <Header />
    <main>
      {children}
    </main>
  </div>
)
PageTemplate.propTypes = {
  children: PropTypes.element.isRequired,
}
export default PageTemplate
