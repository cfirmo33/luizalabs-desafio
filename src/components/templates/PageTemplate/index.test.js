import React from 'react'
import { shallow } from 'enzyme'
import PageTemplate from './index'

describe('PageTemplate', () => {
  it('Deve renderizar corretamente', () => {
    const wrapper = shallow(
      <PageTemplate>
        <h1>Hello World</h1>
      </PageTemplate>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
