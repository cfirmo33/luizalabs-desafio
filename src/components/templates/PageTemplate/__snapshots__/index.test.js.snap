// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTemplate Deve renderizar corretamente 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageTemplate>
    <h1>
        Hello World
    </h1>
</PageTemplate>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(withRouter(Header)) />,
        <main>
          <h1>
                    Hello World
          </h1>
</main>,
        <Connect(ReduxToastr)
          newestOnTop={false}
          position="top-right"
          preventDuplicates={true}
          progressBar={true}
          timeOut={3500}
          transitionIn="fadeIn"
          transitionOut="fadeOut"
/>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Hello World
</h1>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hello World",
          },
          "ref": null,
          "rendered": "Hello World",
          "type": "h1",
        },
        "type": "main",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "newestOnTop": false,
          "position": "top-right",
          "preventDuplicates": true,
          "progressBar": true,
          "timeOut": 3500,
          "transitionIn": "fadeIn",
          "transitionOut": "fadeOut",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(withRouter(Header)) />,
          <main>
            <h1>
                        Hello World
            </h1>
</main>,
          <Connect(ReduxToastr)
            newestOnTop={false}
            position="top-right"
            preventDuplicates={true}
            progressBar={true}
            timeOut={3500}
            transitionIn="fadeIn"
            transitionOut="fadeOut"
/>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Hello World
</h1>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Hello World",
            },
            "ref": null,
            "rendered": "Hello World",
            "type": "h1",
          },
          "type": "main",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "newestOnTop": false,
            "position": "top-right",
            "preventDuplicates": true,
            "progressBar": true,
            "timeOut": 3500,
            "transitionIn": "fadeIn",
            "transitionOut": "fadeOut",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
